keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,931 @@
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+* Announce deprecation and support policy of the existing flag
+* Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+* Address feedback on usage/changed behavior, provided on GitHub issues
+* Deprecate the flag
+-->
+
+#### Alpha
+
+* Initial e2e tests completed and enabled
+* `generation` functaionlity implemented
+* `observedGeneration` functionality implemented behind feature flag
+
+#### Beta
+
+* `generation` and `observedGeneration` functionality have been implemented and running as alpha for at least one release.
+
+#### GA
+
+* No major bugs reported for three months. 
+* User feedback (ideally from at least two distinct users) is green.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+* What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+* What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+API server should be upgraded before Kubelets. Kubelets should be downgraded 
+before the API server. 
+
+Pods that are already running will need to go through at least one update that 
+requires actuation by the kubelet before the API Server populates the `generation`
+field.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+* Does this enhancement involve coordinating behavior in the control plane and nodes?
+* How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+* How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+* Will any other components on the node change? For example, changes to CSI, 
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Previous versions of clients unaware of `generation` functionality would either 
+not set the pod `generation` field (having the effective value of 0) or set it to 
+some custom value, though the latter is unlikely. In either case, the API server
+will ignore whatever value of `generation` is set by the client and will manage 
+`generation` itself (setting it to 1 for newly created pods, or incrementing it 
+for pod updates). 
```

Comment:
What happens today if a client sets `metadata.generation` on pod creation? Does it get stomped to 0 by the API server or left at the custom value?

It looks like there's generic code in place to prevent client-driven changes to the field on update API requests, which is good - https://github.com/kubernetes/kubernetes/blob/a444a5bfedb2807632447bfb8350bd41fd1a3d77/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L122-L127

Reply:
yup I've been able to set pod `metadata.generation` on creation but I cannot update it once set